plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/editors/featuremodel/operations/CreateFeatureAboveOperation.java

```diff
@@ -96,12 +107,76 @@ protected FeatureIDEEvent operation() {
 			featureModel.addFeature(newCompound);
 			featureModel.getStructure().setRoot(newCompound.getStructure());
 		}
+
+		if (newGraphicalFeature.getGraphicalModel().getLayout().getLayoutAlgorithm() == 0) {
+			adjustPositions();
+		}
+
 		return new FeatureIDEEvent(featureModel, EventType.FEATURE_ADD_ABOVE, parent != null ? parent.getFeature() : null, newCompound);
 	}
 
+	/**
+	 * adjusts position of the new feature and its children in manual layout
+	 */
+	private void adjustPositions() {
+		final List<IGraphicalFeature> graphicalChildren = newGraphicalFeature.getGraphicalChildren(true);
+		int minX = graphicalChildren.get(0).getLocation().x;
+		int minY = graphicalChildren.get(0).getLocation().y;
+		int yLocation = graphicalChildren.get(0).getLocation().y;
+		int xLocation = graphicalChildren.get(0).getLocation().x;
+
+		// looks for the leftest x coordinate and the topmost y coordinate of a child
+		for (final IGraphicalFeature child : graphicalChildren) {
+			if (child.getLocation().x < minX) {
+				minX = child.getLocation().x;
+				yLocation = child.getLocation().y;
+			}
+			if (child.getLocation().y < minY) {
+				minY = child.getLocation().y;
+				xLocation = child.getLocation().x;
+			}
+		}
+
+		// decides if the anchor points are at the side or on the top of the rectangle
+		int distance;
+		boolean topDown;
+		if (!newGraphicalFeature.getGraphicalModel().getLayout().verticalLayout()) {
+			newGraphicalFeature.setLocation(new Point(minX, yLocation));
+			topDown = true;
+			distance = topDownDefaultDistance;
+		} else {
+			newGraphicalFeature.setLocation(new Point(xLocation, minY));
+			topDown = false;
+			distance = leftRightDefaultDistance;
+		}
+		this.distance = distance;
+		shiftChildren(graphicalChildren, distance, topDown);
+	}
+
+	/**
+	 * shifts the subtree of the new feature above
+	 *
+	 * @param graphicalChildren children of the new feature
+	 * @param distance distance to shift
+	 * @param topDown previous layout
+	 */
+	private void shiftChildren(List<IGraphicalFeature> graphicalChildren, int distance, boolean topDown) {
+		for (int i = 0; i < graphicalChildren.size(); i++) {
+			shiftChildren(graphicalChildren.get(i).getGraphicalChildren(true), distance, topDown);
+			if (topDown) {
+				graphicalChildren.get(i).setLocation(new Point(graphicalChildren.get(i).getLocation().x, graphicalChildren.get(i).getLocation().y + distance));
+			} else {
```

Comment:
Add a small note that more cases will need to be added in the future for new layouts.